aspnetcore/performance/caching/middleware.md

```diff
@@ -1,15 +1,114 @@
+
+> The Response Caching Middleware only caches 200 (OK) server responses. Any other responses, including [error pages](xref:fundamentals/error-handling), will be ignored by the middleware.
+
+## Options
+The middleware offers two options for controlling response caching.
+
+Option | Default Value
+--- | ---
+UseCaseSensitivePaths | <p>Determines if responses will be cached on case-sensitive paths.</p><p>The default value is `false`.</p>
+MaximumBodySize | <p>The largest cacheable size for the response body in bytes.</p>The default value is `64 * 1024 * 1024` [64 MB (67,108,864 bytes)].</p>
+
+The following example configures the middleware to cache responses smaller than or equal to 1 MB (1,024 bytes) using case-sensitive paths, storing the responses to `/page1` and `/Page1` separately.
+
+```csharp
+services.AddResponseCaching(options =>
+{
+    options.UseCaseSensitivePaths = true;
+    options.MaximumBodySize = 1024;
+});
+```
+
+## `ResponseCache` attribute
+The `ResponseCache` attribute specifies the parameters necessary for setting appropriate headers for response caching. The only parameter of the `ResponseCache` attribute that strictly requires the middleware is `VaryByQueryKeys`, which does not correspond to an actual HTTP header. See [ResponseCache Attribute](response.md#responsecache-attribute) and the [ResponseCacheAttribute Class API Reference](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.aspnetcore.mvc.responsecacheattribute) for usage and a description of the parameters.
+
+## `VaryByQueryKeys` without using the `ResponseCache` attribute
+You can vary response caching with the `VaryByQueryKeys` feature without the `ResponseCache` attribute by using the `ResponseCachingFeature` directly from the `IFeatureCollection` of the `HttpContext`.
+
+```csharp
+var responseCachingFeature = context.HttpContext.Features.Get<IResponseCachingFeature>();
+if (responseCachingFeature != null)
+{
+    responseCachingFeature.VaryByQueryKeys = new[] { "MyKey" };
+}
+```
+
+## HTTP response caching headers
+Response caching by the middleware is configured via your HTTP response headers. The relevant headers are listed below with notes on how they affect caching.
+
+Header | Details
+--- | --- |
+Authorization | <p>The response is not cached if the header exists.</p>
+Cache-Control | <p>The header must be present and its value must explicitly be marked `public`.</p><p>You can control caching with the following parameters:</p><ul><li>max-age</li><li>max-stale</li><li>min-fresh</li><li>must-revalidate</li><li>no-cache</li><li>no-store</li><li>only-if-cached</li><li>private</li><li>public</li><li>s-maxage</li><li>proxy-revalidate</li></ul>
+Pragma | <p>The response is not cached if the value is set to `no-cache`. This header is overridden by the relevant directives in the `Cache-Control` header if present.</p><p>Considered for backward compatibility with HTTP/1.0.</p>
```

Comment:
Sorry I misspoke here, we don't consider the pragma header for responses. We only check the pragma header for requests and it serves the same purpose as `Cache-Control: no-cache` which just prevents a cached response from being served. The middleware will instead contact the server to regenerate the response. As for the documentation here:

"A `Pragma: no-cache` header in the request produces the same effect as `Cache-Control: no-cache`. This header is overridden..."